### PR TITLE
Fall back whole plan when Spark enable ANSI mode.

### PR DESCRIPTION
**Pull request by lviiii**
## What changes were proposed in this pull request?

Fix the issue dividing by zero case mismatched, including "postgreSQL/select_having.sql" and "udf/postgreSQL/udf-select_having.sql".

We find Velox's ANSI Compliance very weak and open an [issue](https://github.com/facebookincubator/velox/issues/3869) to discuss about this. When we enable ANSI mode and run Spark UTs with Velox backend, some cases got mismatched results with Vanilla Spark. So, we are going to take the fallback strategy to avoid this problem.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by JkSelf** [comment]
@lviiii Verified int4.sql in my local env. It can work for me. Can you also add the related failed sqls in this PR?

File: gluten-ut/spark32/src/test/scala/org/apache/spark/sql/GlutenSQLQueryTestSuite.scala [hunk 1]
```diff
@@ -211,10 +211,7 @@ class GlutenSQLQueryTestSuite extends QueryTest with SharedSparkSession with SQL
     "explain-aqe.sql", // explain plan is different
     "explain-cbo.sql", // explain
     "explain.sql", // explain
-    "group-analytics.sql", // wait velox to fix issue 3357
-    // divideByZero case mismatched
-    "postgreSQL/select_having.sql",
```
Comment (lviiii) [inline gluten-ut/spark32/src/test/scala/org/apache/spark/sql/GlutenSQLQueryTestSuite.scala:216]: @JkSelf Failed case are here.

**Review by JkSelf — APPROVED** [review]
LGTM.

**Review by rui-mo — APPROVED** [review]

**Comment by FelixYBW** [comment]
We do we need if spark enable ANSI mode? Can you add more desc in the PR?

**Comment by lviiii** [comment]
> We do we need if spark enable ANSI mode? Can you add more desc in the PR?
[Some UTs](https://github.com/oap-project/gluten/blob/main/gluten-ut/spark32/src/test/scala/org/apache/spark/sql/GlutenSQLQueryTestSuite.scala#L606) enable ANSI mode and got mismatched results with Vanilla Spark, because Velox still don't support ANSI.

**Comment by lviiii** [comment]
I think this update will fall back 90% case with ANSI mode on, because it just fallback expressions.  I've tested another solution to fall back the whole plan by add a transformer rule, like this.
image 
However, the solution cause that most operators will miss the opportunity with native optimizations. How do you think, which solution would you prefer? @FelixYBW @rui-mo

**Comment by FelixYBW** [comment]
> ANSI mode and got mismatched results with Vanilla Spark, because Velox still don't suppor

Is Velox's document mentioned about this? what's their plan?

**Comment by lviiii** [comment]
> > ANSI mode and got mismatched results with Vanilla Spark, because Velox still don't suppor
> 
> Is Velox's document mentioned about this? what's their plan?

No doc mentioned about this, so opened this issue, https://github.com/facebookincubator/velox/issues/3869.

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**